llama.cpp

```diff
@@ -17628,6 +17628,15 @@ static int32_t llama_chat_apply_template_internal(
             }
         }
         // llama2 templates seem to not care about "add_generation_prompt"
+    } else if (tmpl == "phi3" || (tmpl.find("<|assistant|>") != std::string::npos && tmpl.find("<|end|>") != std::string::npos)) {
+        // Phi 3
+        for (auto message : chat) {
+            std::string role(message->role);
+            ss << "<|" << role << "|>\n" << trim(message->content) << "<|end|>\n";
```

Comment:
One small change: None of the template has `message['content'] | trim`, so we shouldn't trim here

```suggestion
            ss << "<|" << role << "|>\n" << message->content << "<|end|>\n";
```

Reply:
Tests fail now obviously.
Will fix soon